### PR TITLE
Added #553

### PR DESCRIPTION
**Pull request by user-promi**

### CODE REVIEW

**Comment by Sangi0** [comment]
https://watch.screencastify.com/v/xn54P4yMFjtkmE9EIDfc

File: api/class-mvx-rest-controller.php [hunk 1]
```diff
@@ -1592,6 +1606,40 @@ public function mvx_list_of_work_board_content() {
             }
         }
 
+        /**
+         * pending vendor deactivation request list
+         */
+        if ($this->mvx_list_of_request_profile_deletion_vendor()->data && !empty($this->mvx_list_of_request_profile_deletion_vendor()->data)) {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: checking missing

File: api/class-mvx-rest-controller.php [hunk 2]
```diff
@@ -1736,7 +1790,71 @@ public function mvx_task_board_icons_triggers($request) {
...
+                foreach ($value as $vendor_id) {
+                    $args = array( 'posts_per_page' => -1, 'author__in' => $vendor_id, 'post_type' => 'product', 'post_status' => 'publish' );
+                    $vendor_query = get_posts($args);
+                    if ($vendor_query) {
+                        foreach ($vendor_query as $product_id){
+                            $update_post_status = array( 'ID' => $product_id->ID, 'post_status' => 'draft' );
+                            wp_update_post($update_post_status);
+                        }
+                    }
+                    $email = WC()->mailer()->emails['WC_Email_Vendor_Account_Deletion_Mail'];
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: checking needed

File: api/class-mvx-rest-controller.php [hunk 3]
```diff
@@ -1736,7 +1790,71 @@ public function mvx_task_board_icons_triggers($request) {
...
+                    $email->trigger($vendor_id);
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: checking needed

File: api/class-mvx-rest-controller.php [hunk 4]
```diff
@@ -2640,11 +2758,43 @@ public function mvx_bulk_todo_pending_product($request) {
                     }
                 }
             }
-        }
+        } elseif (!empty($type) && $type == "requested_profile_deletion") {
+            if ($this->mvx_list_of_request_profile_deletion_vendor()->data && !empty($this->mvx_list_of_request_profile_deletion_vendor()->data)) {
```
Comment (moumitahalder) [inline api/class-mvx-rest-controller.php]: checking

File: mvx-modules/src/components/admin/classes/class-mvx-workboard-section.js [hunk 5]
```diff
@@ -101,8 +105,79 @@ class MVXworkboard extends Component {
...
+				method: 'post',
+				url: `${appLocalizer.apiUrl}/mvx_module/v1/task_board_icons_triggers`,
+				data: {
+					value: this.state.selected_ids,
+					key: this.state.selected_option,
+				},
+			}).then((response) => {
+				if ( response && response.data ) {
+					this.setState({
+						list_of_work_board_content: response.data,
```
Comment (moumitahalder) [inline mvx-modules/src/components/admin/classes/class-mvx-workboard-section.js:136]: checking

**Review by moumitahalder — CHANGES_REQUESTED** [review]

**Comment by moumitahalder** [comment]
few reviews given. please go through the code thoroughly

**Pull request closed without merging**